registry.go

```diff
@@ -357,6 +357,33 @@ func (registry *registryImpl) Poll() *metrics_pb.MetricsMessage {
 	return (*metrics_pb.MetricsMessage)(builder)
 }
 
+func (registry *registryImpl) pollAppend(builder *messageBuilder) *metrics_pb.MetricsMessage {
+	if len(builder.IntervalCounters) == 0 && len(builder.UsageCounters) == 0 && registry.metricMap.Count() == 0 {
```

Comment:
maybe could be a separate function? as is just a teeny bit hard to read/grok? meh?

Reply:
Moved to usage registry impl